telegram-bot-api/Client.cpp

```diff
@@ -6008,9 +6289,9 @@ td::int64 Client::get_message_id(const Query *query, Slice field_name) {
   }
 
   int arg = td::to_integer<int32>(s_arg);
-  if (arg < 0) {
-    return 0;
-  }
+  // if (arg < 0) {
```

Comment:
Why did you remove this check?

Reply:
Scheduled messages have negative message ids. Therefore this check if the message_id is negative breaks scheduled messages.

Reply:
Ok, but this could possibly break other methods that expect the message_id to be positive, especially with bots that can't schedule messages, or methods that should work with non scheduled messages, adding a parameter to allow negative numbers would be fine for me

Reply:
The check basically moved to as_tdlib_message_id. There it uses the old conversion (20 bit shift) only for positive message_ids. It still has the same check for positive numbers and an other behaviour for negative ones.

Reply:
arg < 0 && is_bot

Reply:
Okay, I will add that.